EXT_BeamPhysics.md

```diff
@@ -0,0 +1,346 @@
+    - Attributes:
+
+- `spin/`
+    - Type: Optional 3-vector *(real)*
+    - Description: Particle spin.
+    - Components: (`x`, `y`, `z`) or (`r`, `theta`, `phi`).
+
+- `time/`
+    - Type: Optional *(real)*
+    - Description: Time relative to `timeOffset`. That is, absolute time = `time + timeOffset`.
+
+- `timeOffset/`
+    - Type: Optional *(real)*
+    - Description: Base time from which `time` is measured. That is, absolute time = `time + timeOffset`. Assumed zero if not present. Some programs will use the `timeOffset` to store the **reference time** in which case `time` will then be the deviation from the reference.
+
+- `velocity/`
+  - Type: Optional 3-vector *(real)*
+  - Description: (`x`, `y`, `z`) velocity vector. Meant to be used for photons where using `momentum` is not appropriate.
+
+- `weight/`
+    - Type: Optional *(real)*
+    - Description: If macro-particles are being simulated, the `weight` is the total charge of a macro-particle. This is proportional to the number of particles that a macro-particle represents. Also see `charge`.
+
+
+Non Per-Particle Records of the `Particle Group`
+-----------------------------------------------------
+
+The following possible records of the `Particle Group` are for specifying properties of the entire group of particles.
+
+- `phaseSpaceFirstOrderMoment/`
+  - Type: Optional 6-vector *(real)*
+  - Description: First order beam moments of `(x, px, y, py, z, pz)`.
+
+- `phaseSpaceSecondOrderMoment/`
+  - Type: Optional 6x6-matrix *(real)*
+  - Description: Second order beam moments of `(x, px, y, py, z, pz)`.
+
+Particle Record Dataset Attributes
+----------------------------------
+
+The following attributes can be used with any dataset:
+
+- `minValue`:
+  - Type: Optional *(real)*
+  - Description: Minimum of the data.
+
+- `maxValue`:
+  - Type: Optional *(real)*
+  - Description: Maximum of the data.
+
+External Mesh Fields Groups
+===========================
+
+The **external mesh field group** is a group for specifying electric and/or magnetic fields, due to a lattice element, at regularly spaced grid points. For example, the fields due to an RF cavity or the fields due to a magnet. Multiple **external mesh field groups** can be defined in a file. The path for a **external mesh field group** is given by the **externalFieldPath** in the file root group:
+- `externalFieldPath`
+  - Type: Required if there are external mesh field group(s) *(string)*
+  - Description: Base path to the external mesh field groups. Use the **%T** construct if there are multiple meshes present.
+  - Example: `/ExternalFieldMesh/%T/`. Base paths to the external fields group, in this case, would be `/ExternalFieldMesh/1/`, etc.
+  - Example: `/ExternalFieldMesh/`. In this case there is only one external fields group.
+
+Notes
+-----
+
+- AC fields can be described using complex numbers. The actual field is the real part of
+
+    Z &ast; Exp[-2 &ast; pi &ast; I &ast; (f &ast; t + phi)]
```

Comment:
verbatim with triple `'s probably works well, too:
```
    Z * Exp[-2 * pi * I * (f * t + phi)]
```

Reply:
Will be in the next push...